zerver/views/streams.py

```diff
@@ -224,10 +235,15 @@ def update_stream_backend(
             Stream.STREAM_POST_POLICY_TYPES), default=None),
         history_public_to_subscribers: Optional[bool]=REQ(validator=check_bool, default=None),
         new_name: Optional[str]=REQ(validator=check_string, default=None),
+        message_retention_days: Optional[Union[int, str]]=REQ(validator=check_string_or_int, default=None),
 ) -> HttpResponse:
     # We allow realm administrators to to update the stream name and
     # description even for private streams.
     stream = access_stream_for_delete_or_update(user_profile, stream_id)
+    if message_retention_days is not None:
+        message_retention_days_value = parse_message_retention_days(message_retention_days)
+        do_change_stream_message_retention_days(stream, message_retention_days_value)
```

Comment:
We should enforce the user is a realm owner here (and raise the right exception!), not just an admin, right?

Reply:
Maybe worth doing in `parse_message_retention_days`, and having that take the UserProfile as an argument.

We also need to check `plan_type`; @hackerkid can help with that if you're not sure how.

Reply:
During CZO discussion, you told that we will allow admins to set this property.
If we want to change, we can do owners.